docs/docs/guides/ecosystem/huggingface.md

```diff
@@ -0,0 +1,116 @@
+---
+sidebar_position: 1
+hide_table_of_contents: true
+---
+
+
+# Huggingface
+
+Recording traces of LLM applications in a centralized database is crucial during both development and production phases. These traces are invaluable for debugging and for creating a dataset of challenging examples to test and enhance your application.
+
+You can use models from the Hugging Face ecosystem with Weave to keep track of all your interactions and generate insights.
+
+## Using a models locally
+
+You have fine-tuned a model from the Hugging Face HUB a now you want to interact with it. You can start tracing your model by wrapping the model call with a `weave.op()`:
```

Comment:
@tcapelle I just had a look at the other ecosystem docs (OpenAI, Anthropic, Mistral) and for them the same main issue applies as for this HF pipeline - we init a new model object on every predict call. 

If the model inference happens in the cloud re-creating the model object comes down to re-creating a client object that is cheap to re-init (that's why for OpenAI, etc. it didn't make a big difference) but if the inference happens locally then the model object can't be re-init on every inference call (downloading, loading in memory, etc. would take time). 

Imo would make sense to create dedicated page in the docs about model init best practices (regarding the constructor) and related to that for weave objects and serialization. And then from the HF page (and also the other ecosystem pages) we could link to them - what do you think?

Reply:
agree